pandas/_libs/reshape.pyx

```diff
@@ -91,3 +93,59 @@ unstack_int64 = unstack["int64_t"]
 unstack_float32 = unstack["float32_t"]
 unstack_float64 = unstack["float64_t"]
 unstack_object = unstack["object"]
+
+
+@cython.wraparound(False)
+@cython.boundscheck(False)
+def explode(ndarray[object] values):
+    """
+    transform array list-likes to long form
+    preserve non-list entries
+
+    Parameters
+    ----------
+    values : object ndarray
+
+    Returns
+    -------
+    tuple(values, counts)
+    """
+    cdef:
+        Py_ssize_t i, j, count, n
+        object v
+        ndarray[object] result
+        ndarray[uint8_t] counts
+
+    # find the resulting len
+    n = len(values)
+    counts = numpy.zeros(n, dtype='uint8')
```

Comment:
Should we use a larger dtype than 'uint8' or is there assumption that we won't be exploding more than 255 elements for a given index?  Looks like there's a seg fault when trying to explode 256+ elements:
```python
In [2]: pd.Series([range(255)]).explode()
Out[2]: 
0      0
0      1
0      2
0      3
0      4
    ... 
0    250
0    251
0    252
0    253
0    254
Length: 255, dtype: object

In [3]: pd.Series([range(256)]).explode()
Segmentation fault (core dumped)
```

Reply:
good point.